src/matUtils/merge.cpp

```diff
@@ -0,0 +1,131 @@
+ //Load mat files
+ //uncondense leaves
+ #include "merge.hpp"
+ po::variables_map parse_merge_command(po::parsed_options parsed) {
+    uint32_t num_cores = tbb::task_scheduler_init::default_num_threads();
+    std::string num_threads_message = "Number of threads to use when possible [DEFAULT uses all available cores, " + std::to_string(num_cores) + " detected on this machine]";
+
+    po::variables_map vm;
+    po::options_description merge_desc("merge options");
+    merge_desc.add_options()
+        ("input-mat-1,ione", po::value<std::string>()->required(),
+         "Input mutation-annotated tree file [REQUIRED]. If only this argument is set, print the count of samples and nodes in the tree.")
+        ("input-mat-2,itwo", po::value<std::string>()->required(),
+         "Input mutation-annotated tree file [REQUIRED]. If only this argument is set, print the count of samples and nodes in the tree.")
+        ("output-mat,o", po::value<std::string>()->required(),
+        "Write output files to the target directory. Default is current directory.");
+    std::vector<std::string> opts = po::collect_unrecognized(parsed.options, po::include_positional);
+    opts.erase(opts.begin());
+
+    // Run the parser, with try/catch for help
+    try{
+        po::store(po::command_line_parser(opts)
+                  .options(merge_desc)
+                  .run(), vm);
+        po::notify(vm);
+    }
+    catch(std::exception &e){
+        std::cerr << merge_desc << std::endl;
+        // Return with error code 1 unless the user specifies help
+        if (vm.count("help"))
+            exit(0);
+        else
+            exit(1);
+    }
+    return vm;
+}  
+
+bool consistent(MAT::Tree A, MAT::Tree B){
+    //vectors of all leaves in both input trees
+    std::vector<std::string> A_leaves = A.get_leaves_ids();
+    std::vector<std::string> B_leaves = B.get_leaves_ids();
+    //creates a vector of common_leaves between two input trees
+    std::vector<std::string> common_leaves(std::max(A_leaves.size(), B_leaves.size()));
+    set_intersection(A_leaves.begin(),A_leaves.end(),B_leaves.begin(),B_leaves.end(), std::back_inserter(common_leaves));
+    //creates two subtrees using the common_leaves
+    auto Asub = MAT::get_subtree(A, common_leaves);
+    auto Bsub = MAT::get_subtree(B, common_leaves);
+    
+    auto Adfs = Asub.depth_first_expansion();
+    auto Bdfs = Bsub.depth_first_expansion();
+
+    if (Adfs.size() != Bdfs.size()){
+        return false;
+    }
+    
+    for (int i = 0; i < Adfs.size(); i++){
```

Comment:
@dtorvi You should try to parallelize this for loop. See `tbb::parallel_for` in usher.cpp for examples.